tsdb/index/tsi1/cache.go

```diff
@@ -198,3 +218,82 @@ type seriesIDCacheElement struct {
 	value       string
 	SeriesIDSet *tsdb.SeriesIDSet
 }
+
+type cacheTracker struct {
```

Comment:
From here down is pretty much copy-paste. My main addition was adding `if t == nil` checks to many of the methods, to support running without metrics-tracking (an option in the beta, not sure if we want to keep it).

Reply:
Need to add a comment explaining why this struct exists